drivers/scheduler/rke/rke.go

```diff
@@ -1,28 +1,236 @@
+	_ "github.com/rancher/norman/clientbase"
+	rancherClientBase "github.com/rancher/norman/clientbase"
+	rancherClient "github.com/rancher/rancher/pkg/client/generated/management/v3"
+	"io/ioutil"
+	"net/http"
+	"strings"
 )
 
 const (
-	// SchedName is the name of the kubernetes scheduler driver implementation
-	SchedName = "rke"
-	// SystemdSchedServiceName is the name of the system service resposible for scheduling
-	SystemdSchedServiceName = "kubelet"
+	// scheduleName is the name of the kubernetes scheduler driver implementation
+	scheduleName = "rke"
+	// SystemdScheduleServiceName is the name of the system service responsible for scheduling
+	SystemdScheduleServiceName = "kubelet"
+	// PxLabelNameKey is key for map
+	PxLabelNameKey = "name"
+	// PxLabelValue portworx pod label
+	PxLabelValue = "portworx"
 )
 
-type rke struct {
+var RancherClusterParametersValue *RancherClusterParameters
+
+type Rancher struct {
 	kube.K8s
+	client *rancherClient.Client
+}
+
+type RancherClusterParameters struct {
+	Token     string
+	Endpoint  string
+	AccessKey string
+	SecretKey string
+}
+
+// String returns the string name of this driver.
+func (k *Rancher) String() string {
+	return scheduleName
+}
+
+// Init Initialize the driver
+func (k *Rancher) Init(scheduleOpts scheduler.InitOptions) error {
+	var err error
+	nodes, err := core.Instance().GetNodes()
+	if err != nil {
+		return err
+	}
+	for _, n := range nodes.Items {
+		if err = k.AddNewNode(n); err != nil {
+			return err
+		}
+	}
+	// Update node PxPodRestartCount during init
+	namespace, err := k.GetAutopilotNamespace()
+	if err != nil {
+		log.Fatalf(fmt.Sprintf("%v", err))
+	}
+	pxLabel := make(map[string]string)
+	pxLabel[PxLabelNameKey] = PxLabelValue
+	pxPodRestartCountMap, err := k.GetPodsRestartCount(namespace, pxLabel)
+	if err != nil {
+		log.Fatalf(fmt.Sprintf("%v", err))
+	}
+	for pod, value := range pxPodRestartCountMap {
+		n, err := node.GetNodeByIP(pod.Status.HostIP)
+		if err != nil {
+			log.Fatalf(fmt.Sprintf("%v", err))
+		}
+		n.PxPodRestartCount = value
+	}
+	k.SpecFactory, err = spec.NewFactory(scheduleOpts.SpecDir, scheduleOpts.VolDriverName, k)
+	if err != nil {
+		return err
+	}
+	err, RancherClusterParametersValue = k.GetRancherClusterParametersValue()
+	if err != nil {
+		return err
+	}
+	rancherClientOpts := rancherClientBase.ClientOpts{
+		URL:       RancherClusterParametersValue.Endpoint,
+		TokenKey:  RancherClusterParametersValue.Token,
+		AccessKey: RancherClusterParametersValue.AccessKey,
+		SecretKey: RancherClusterParametersValue.SecretKey,
+		Insecure:  true,
+	}
+	k.client, err = rancherClient.NewClient(&rancherClientOpts)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// GetRancherClusterParametersValue returns the rancher token,endpoint,secret key, access key
+func (k *Rancher) GetRancherClusterParametersValue() (error, *RancherClusterParameters) {
+	var data map[string]interface{}
+	var var1 RancherClusterParameters
```

Comment:
Please provide more meaningful variable names got both of these

Reply:
done